beard_server/modules/text/views.py

```diff
@@ -51,35 +48,60 @@
 def available_methods():
     """List of available methods to call as the part of API."""
     available = {
-        "clusters": {
-            "path": "/api/text/phonetic_block",
-            "method": "[GET]",
+        "phonetic_blocks": {
+            "path": "/api/text/phonetic_blocks",
+            "method": "[POST]",
             "example": {
-                "request": "/api/text/phonetic_block?full_name=John%20Smith",
-                "response": {"phonetic_block": "SNATHj"}
+                "body": {
+                    "full_names": [
+                        "John Smith"
+                    ]
+                },
+                "header": {
+                    "Accept": "application/json",
+                    "Content-Type": "application/json"
+                },
+                "response": {
+                    "phonetic_blocks": {
+                        "John Smith": "SNATHj"
+                    }
+                }
             }
         }
     }
 
     return jsonify(available)
 
 
-@blueprint.route('/phonetic_block', methods=['GET'])
-def get_phonetic_block():
+@blueprint.route('/phonetic_blocks', methods=['POST'])
+def get_phonetic_blocks():
     """Create phonetic block from a given full name using nysiis algorithm."""
+    full_names_data = request.get_json(silent=True)
+
+    # Unpack the body of the POST request.
     try:
-        name = {'author_name': request.values.get('full_name', type=unicode)}
-
-        signature_block = block_phonetic(
-            np.array([name], dtype=np.object).reshape(-1, 1),
-            threshold=0,
-            phonetic_algorithm='nysiis'
-        )[0]
-    except IndexError:
-        # Most likely a malformed author name.
-        abort(420)
-    except TypeError:
-        # Missing full_name argument.
+        full_names = full_names_data['full_names']
+    except KeyError:
+        # Missing data.
         abort(400)
 
-    return jsonify({"phonetic_block": signature_block})
+    try:
+        response = phonetic_blocks(full_names)
+    except TypeError:
+        # Malformed author's name.
+        full_names_trimmed = []
+
+        for full_name in full_names:
+            try:
+                # Most probably a given name cannot be converted to unicode.
+                if not isinstance(full_name, unicode):
```

Comment:
I don't understand: either `full_name` is a `unicode` or it is not. Isn't this just depending simply on the `get_json` line above?